web/models/coherence/user.ex

```diff
@@ -88,50 +102,51 @@ defmodule Vae.User do
     end
   end
 
+  def build_api_calls do
+    [%{
+      url: "https://api.emploi-store.fr/partenaire/peconnect-coordonnees/v1/coordonnees",
+      is_data_missing: &(is_nil(&1.postal_code)),
+      data_map: &__MODULE__.coordonnees_api_map/1
+    }, %{
+      url: "https://api.emploi-store.fr/partenaire/peconnect-competences/v2/competences",
+      is_data_missing: &(length(&1.skills) == 0),
+      data_map: fn data -> %{skills: Enum.map(data, &Skill.competences_api_map/1)}
+      end
+    }, %{
+      url: "https://api.emploi-store.fr/partenaire/peconnect-experiences/v1/experiences",
+      is_data_missing: &(length(&1.experiences) == 0),
+      data_map: fn data -> %{experiences: Enum.map(data, &Experience.experiences_api_map/1)}
+      end
+    }] ++ Enum.map(1..5, fn i ->
+      # Since API called is limited to a 2 years interval, we need to fetch it 5 times to get 10 years
+      start_date = Timex.shift(Timex.today, years: -2 * i, days: 1)
+      end_date = Timex.shift(Timex.today, years: -2 * (i-1))
+
+      %{
+        url:
+          "https://api.emploi-store.fr/partenaire/peconnect-experiencesprofessionellesdeclareesparlemployeur/v1/contrats?dateDebutPeriode=#{Timex.format!(start_date, "{YYYY}{0M}{0D}")}&dateFinPeriode=#{Timex.format!(end_date, "{YYYY}{0M}{0D}")}",
+        is_data_missing: fn user -> length(Enum.filter(user.proven_experiences, fn exp -> Timex.between?(exp.start_date, start_date, end_date) end)) == 0 end,
+        data_map: fn data ->
+          %{
+            proven_experiences:
+              Enum.filter(Enum.map(
+                data["contrats"],
+                &ProvenExperience.experiencesprofessionellesdeclareesparlemployeur_api_map/1
+                # To make sure `is_data_missing` works properly,
```

Comment:
Ca c'est le hic de l'API PE qui dit:

>  La recherche se fait sur tous les contrats ayant été en cours dans la période (qu’ils soient terminés ou non)

ce qui  fait qu'une même expérience peut sortir dans 2 requêtes différentes, et donc renvoyer `false` is `is_data_missing` alors qu'on n'a pas encore crawlé la période et qu'il en manque la majorité...

Reply:
Le plus propre serait de ne faire qu'une requête, apparemment ça n'est pas perdu !